scripts/quests/otherAreas/A_Generous_General.lua

```diff
@@ -0,0 +1,290 @@
+-----------------------------------
+-- A Generous General
+-----------------------------------
+-- Log ID: 4, Quest ID: 109
+-----------------------------------
+-- Oldton zone !pos 566 -12 685 106
+-- Faulpie     !gotoid 17719379
+-- iron box    !pos -141 7 157 11
+-----------------------------------
+local oldtonID = zones[xi.zone.OLDTON_MOVALPOLOS]
+local southernSanDoriaID = zones[xi.zone.SOUTHERN_SAN_DORIA]
+-----------------------------------
+local quest = Quest:new(xi.questLog.OTHER_AREAS, xi.quest.id.otherAreas.A_GENEROUS_GENERAL)
+
+local spawnedMobs =
+{
+    oldtonID.mob.GENEROUS_GENERAL_OFFSET,
+    oldtonID.mob.GENEROUS_GENERAL_OFFSET + 1,
+    oldtonID.mob.GENEROUS_GENERAL_OFFSET + 2,
+    oldtonID.mob.GENEROUS_GENERAL_OFFSET + 3,
+    oldtonID.mob.GENEROUS_GENERAL_OFFSET + 4,
+    oldtonID.mob.GENEROUS_GENERAL_OFFSET + 5,
+    oldtonID.mob.GENEROUS_GENERAL_OFFSET + 6,
+}
+
+quest.reward =
+{
+    item  = xi.item.CHOPLIXS_COIF,
+    title = xi.title.MOBLIN_KINSMAN,
+}
+
+quest.sections =
+{
+    {
+        check = function(player, status, vars)
+            return status == xi.questStatus.QUEST_AVAILABLE and
+                player:getQuestStatus(xi.questLog.OTHER_AREAS, xi.quest.id.otherAreas.AN_AFFABLE_ADAMANTKING) ~= xi.questStatus.QUEST_ACCEPTED and
+                player:getQuestStatus(xi.questLog.OTHER_AREAS, xi.quest.id.otherAreas.AN_UNDERSTANDING_OVERLORD) ~= xi.questStatus.QUEST_ACCEPTED and
+                player:getQuestStatus(xi.questLog.OTHER_AREAS, xi.quest.id.otherAreas.A_MORAL_MANIFEST) ~= xi.questStatus.QUEST_ACCEPTED and
+                player:getVar('BstHeadGearQuest_Conquest') < NextConquestTally() and
+                not player:hasItem(xi.item.CHOPLIXS_COIF) and
+                player:getMainLvl() >= 60
+        end,
+
+        [xi.zone.OLDTON_MOVALPOLOS] =
+            {
+            onZoneIn =
+            {
+                function(player, prevZone)
+                    local pos = player:getPos()
+                    if
+                        pos.x >= -323.0 and
```

Comment:
These conditions are checked multiple times.  Should move these coordinate checks to their own function for reusability